homeassistant/components/velbus/__init__.py

```diff
@@ -42,68 +53,88 @@ async def async_setup(hass, config):
     return True
 
 
+async def velbus_connect_task(
+    controller: Velbus, hass: HomeAssistant, entry_id: str
+) -> None:
+    """Task to offload the long running connect."""
+    # create notification
+    persistent_notification.async_create(
+        hass,
+        "Velbus is scanning the bus, this can take some time.",
+        "Velbus Starting",
+        f"{DOMAIN}-{entry_id}",
+    )
+    # connect
+    await controller.connect()
+    # clear notification
+    persistent_notification.async_dismiss(hass, f"{DOMAIN}-{entry_id}")
+
+
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Establish connection with velbus."""
     hass.data.setdefault(DOMAIN, {})
 
     controller = Velbus(entry.data[CONF_PORT])
-    hass.data[DOMAIN][entry.entry_id] = controller
-    await controller.connect()
+    hass.data[DOMAIN][entry.entry_id] = {}
+    hass.data[DOMAIN][entry.entry_id]["cntrl"] = controller
+    hass.data[DOMAIN][entry.entry_id]["tsk"] = hass.async_create_task(
+        velbus_connect_task(controller, hass, entry.entry_id)
+    )
 
     hass.config_entries.async_setup_platforms(entry, PLATFORMS)
 
-    if hass.services.has_service(DOMAIN, "scan"):
+    if hass.services.has_service(DOMAIN, SERVICE_SCAN):
         return True
 
-    async def get_entry_id(interface: str):
+    def get_entry_id(interface: str) -> str | None:
         for entry in hass.config_entries.async_entries(DOMAIN):
             if "port" in entry.data and entry.data["port"] == interface:
                 return entry.entry_id
         _LOGGER.warning("Can not find the config entry for: %s", interface)
         return None
 
+    def check_entry_id(interface: str):
+        for entry in hass.config_entries.async_entries(DOMAIN):
+            if "port" in entry.data and entry.data["port"] == interface:
+                return interface
+        raise vol.Invalid(
+            "The interface provided is not defined as a port in a Velbus integration"
+        )
+
     async def scan(call):
-        entry_id = await get_entry_id(call.data["interface"])
+        entry_id = get_entry_id(call.data[CONF_INTERFACE])
         if not entry_id:
             return
-        await hass.data[DOMAIN][entry_id].scan()
+        await hass.data[DOMAIN][entry_id]["cntrl"].scan()
 
     hass.services.async_register(
         DOMAIN,
-        "scan",
+        SERVICE_SCAN,
         scan,
-        vol.Schema(
-            {
-                vol.Required("interface"): cv.string,
-            }
-        ),
+        vol.Schema({vol.Required(CONF_INTERFACE): vol.All(check_entry_id, cv.string)}),
```

Comment:
Start with the string validation.
```suggestion
        vol.Schema({vol.Required(CONF_INTERFACE): vol.All(cv.string, check_entry_id)}),
```